demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -215,14 +409,454 @@ static bool getS3ObjectFileSize( size_t * pFileSize,
+
+static void lowercaseHexEncode( const char * pInputStr,
+                                size_t inputStrLen,
+                                char * pHexOutput )
+{
+    static const char digitArr[] = "0123456789abcdef";
+    char * hex = pHexOutput;
+    size_t i = 0U;
+
+    assert( pInputStr != NULL );
+    assert( inputStrLen > 0 );
+    assert( pHexOutput != NULL );
+
+    for( i = 0; i < inputStrLen; i++ )
+    {
+        *hex = digitArr[ ( pInputStr[ i ] & 0xF0 ) >> 4 ];
+        hex++;
+        *hex = digitArr[ ( pInputStr[ i ] & 0x0F ) ];
+        hex++;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t sha256( const char * pInput,
+                       size_t ilen,
+                       char * pOutput )
+{
+    return mbedtls_sha256_ret( ( const unsigned char * ) pInput, ilen, ( unsigned char * ) pOutput, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t sha256Init( void * hashContext )
+{
+    mbedtls_sha256_init( ( mbedtls_sha256_context * ) hashContext );
+    return mbedtls_sha256_starts_ret( ( mbedtls_sha256_context * ) hashContext, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t sha256Update( void * hashContext,
+                             const char * pInput,
+                             size_t inputLen )
+{
+    return mbedtls_sha256_update_ret( ( mbedtls_sha256_context * ) hashContext,
+                                      ( const unsigned char * ) pInput,
+                                      inputLen );
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t sha256Final( void * hashContext,
+                            char * pOutput,
+                            size_t outputLen )
+{
+    assert( outputLen >= SHA256_HASH_DIGEST_LENGTH );
+
+    ( void ) outputLen;
+
+    return mbedtls_sha256_finish_ret( ( mbedtls_sha256_context * ) hashContext,
+                                      ( unsigned char * ) pOutput );
+}
+
+static void getHeaderStartLocFromHttpRequest( HTTPRequestHeaders_t requestHeaders,
+                                              char ** pStartHeaderLoc,
+                                              size_t * pHeadersDataLen )
+{
+    size_t headerLen = requestHeaders.headersLen;
+    char * pHeaders = ( char * ) requestHeaders.pBuffer;
+    bool newLineFound = false;
+
+    assert( pStartHeaderLoc != NULL );
+    assert( pHeadersDataLen != NULL );
+
+    while( headerLen >= 2 && 0 != strncmp( pHeaders, "\r\n", strlen( "\r\n" ) ) )
+    {
+        pHeaders++;
+        headerLen--;
+    }
+
+    if( ( headerLen >= 2 ) && ( 0 == strncmp( pHeaders, "\r\n", strlen( "\r\n" ) ) ) )
+    {
+        newLineFound = true;
+    }
+
+    assert( newLineFound != false );
+
+    if( newLineFound == false )
+    {
+        LogError( ( "HTTP Headers missing \"\\r\\n\" at the end of initial HTTP headers added using coreHTTP Library." ) );
+    }
```

Comment:
```suggestion
    while( headerLen >= 2 )
    {
         if( 0 != strncmp( pHeaders, "\r\n", strlen( "\r\n" ) )
         {
              newLineFound = true;
              break;
         }
         
        pHeaders++;
        headerLen--;
    }

    if( newLineFound == false )
    {
        LogError( ( "Failed to find starting location of HTTP headers in HTTP request: \"\\r\\n\" missing before start of HTTP headers." ) );
    }
    
    assert( newLineFound != false );
```

Reply:
The condition  should be this ( 0 == strncmp( pHeaders, "\r\n", strlen( "\r\n" ) ). I have addressed it.